### PR TITLE
util/chunk: fix incorrect result when set duration to `MutRow`

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

The duration column's len in `MutRow` is 16 while it's 8 in chunk.
So it will meet incorrect result when you append multiple rows from `MutRow` to chunk then read them.

### What is changed and how it works?

This pr only fixes the column len.
It may be refactored in future pr.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note `fix incorrect result when union scan meets duration`

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8725)

### CODE REVIEW

File: util/chunk/mutrow_test.go [hunk 1]
```diff
@@ -78,6 +78,13 @@ func (s *testChunkSuite) TestMutRow(c *check.C) {
 	row = mutRow.ToRow()
 	c.Assert(row.GetJSON(0), check.DeepEquals, j)
 	c.Assert(row.GetTime(1), check.DeepEquals, t)
+
+	retTypes := []*types.FieldType{types.NewFieldType(mysql.TypeDuration)}
+	chk := New(retTypes, 1, 1)
+	chk.AppendMyDecimal(0, types.NewDecFromFloatForTest(1.123))
```
Comment (XuHuaiyu) [inline util/chunk/mutrow_test.go]: Why we append a decimal to a Duration column?

**Comment by zz-jason** [comment]
Seems we forget to modify the implementation in MutRow when we refactor the `Duration` data representation in `Chunk`.

Is it safe to discard the `Fsp` info in the original `Duration` value? In other places, that info is stored in the executor's schema, so it can be removed when appended into `Chunk`, maybe we have to check this.

File: util/chunk/mutrow_test.go (hunk 1 above)
Comment (winoros) [reply util/chunk/mutrow_test.go]: Oh..you're right.. I append the wrong thing...

**Comment by winoros** [comment]
@XuHuaiyu He means that delete `Fsp` from struct `Duration`.

There're also other ways:
1. Refactor `SetXXX` of `MutRow`, use `chunk`'s `AppendXXX` directly.
2. Add unit test to check that the appended bytes value in `MutRow` is the same with `Chunk`'s append method.

**Pull request merged**